### PR TITLE
feat: adding examples as part of integration tests and misc fixes

### PR DESCRIPTION
**Pull request by jpwsutton**
---
## Status
**READY**

## Description
This PR makes a few changes to the automated tests in ibm-generative-ai
 - Adds the example programs as part of the integration test suite
 - Modifies a few examples to prevent them from running indefinitely / a long time
 - Pushes Code coverage to coveralls
 - Refactors extension tests into their own directory and removes the need for individual pytest markings. Now all extension tests use the `extension` mark.

## Impacted Areas in Library
- GitHub Workflows
- Extension tests
- Integration Tests

## Which issue(s) does this pull-request fix?
Closes: #22 

## Any special notes for your reviewer?

---

## Checklist
- [x] Automated tests exist
- [x] Updated Package Requirements (if required, and with maintainers' approval)
- [x] Local unit tests performed
- [x] Documentation exists [link]()
- [x] Local pre-commit hooks performed
- [x] Desired commit message set as PR title and description set above
- [x] Link to relevant GitHub issue provided

### CODE REVIEW

**Review by vedem1192 — APPROVED** [review]
looks good!

File: tests/integration/test_examples.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+def idfn(val):
+    script_path = pathlib.Path(val)
+    return script_path.name
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("script", scripts, ids=idfn)
+def test_example_execution(script):
+    logger.info(f"Executing Example scipt: {script}")
+    runpy.run_path(script)
```
[the remark this answers is not in the document]
Comment (jpwsutton) [reply tests/integration/test_examples.py]: Thanks @mirianfsilva looks like the newer python versions are stricter with Path objects. I've updated to use `str(script)`.

**Review by mirianfsilva — APPROVED** [review]
Now LGTM! :)

File: examples/user/country-capital-qa.py [hunk 2]
```diff
@@ -29,8 +30,10 @@
 creds = Credentials(api_key, api_endpoint)
 model = Model(ModelType.FLAN_UL2, params=params, credentials=creds)
 
+prompt_path = pathlib.Path(__file__, "..", "prompts", "Country-Capital-Factual-QA").resolve()
```
Comment (onkarbhardwaj) [inline examples/user/country-capital-qa.py:33]: 👍

**Review by onkarbhardwaj — APPROVED** [review]

**Comment by onkarbhardwaj** [comment]
Approved and merging!

**Pull request merged**